File: week18/sangmin/problem3/Problem3.java
package week18.sangmin.problem3;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

class Solution {

    String regex = "[1-9]+";
    Pattern pattern = Pattern.compile(regex);

    boolean isPrime(long num){
        if(num == 1) return false;
        if(num == 2) return true;

        for(long i = 2; i <= Math.sqrt(num); i++){
            if(num % i == 0) return false;
        }

        return true;
    }

    public int solution(int n, int k) {
        int answer = 0;
        StringBuilder sb = new StringBuilder();
        while (n > 0){
            sb.append(n % k);
            n /= k;
        }
        Matcher matcher = pattern.matcher(sb.reverse().toString());
        while (matcher.find()){
            if(isPrime(Long.parseLong(matcher.group()))) answer++;
        }
        return answer;
    }
}

public class Problem3 {
    public static void main(String[] args) {
        new Solution().solution(437674, 3);
    }
}

File: week18/juyoung/problem3/problem3.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <cmath>
#include <iostream>

using namespace std;

bool isPrime(long long n) {
	if (n <= 1) return false;
	if (n == 2) return true;
	for (long long i = 3; i <= sqrt(n); i++) {
		if (n%i == 0) return false;
	}
	return true;
}

int solution(int n, int k) {
	int answer = 0;

	// n을 k진수로 바꾸기
	string number = "";
	char c = 0;
	while (n > 0) {
		c = n % k + '0';
		number += c;
		n /= k;
	}
	reverse(number.begin(), number.end());

	// number에서 10진수 소수 찾기
	string temp = "";
	for (int i = 0; i < number.length(); i++) {
		if (number[i] == '0') {
			if (temp != "" && isPrime(stoll(temp)))
				answer++;
			temp = "";
			continue;
		}
		temp += number[i];
	}

	if (temp != "")
		if (isPrime(stoll(temp)))
			answer++;

	return answer;
}

int main() {
	int n, k, answer;

	cin >> n >> k;
	answer = solution(n, k);
	cout << answer;

	return 0;
}